warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,329 @@
+        String startDate = DateHelper.format(configuration.getBeginDate().getTime());
+        String endDate = DateHelper.format(configuration.getEndDate().getTime());
+        
+        YearMonthDay.Bounds bounds = new YearMonthDay.Bounds(startDate, false, endDate, false);
+        
+        TreeSet<YearMonthDay> queryDates = new TreeSet<>();
+        for (ValueIndexHole valueIndexHole : configuration.getValueIndexHoles()) {
+            addDatesToSet(bounds, queryDates, valueIndexHole.getStartDate());
+            addDatesToSet(bounds, queryDates, valueIndexHole.getEndDate());
+        }
+        
+        for (FieldIndexHole fieldIndexHole : configuration.getFieldIndexHoles()) {
+            // TODO remove comparison below. calculateFieldHoles needs to be fixed.
+            if (fieldIndexHole.getStartDate().compareTo(fieldIndexHole.getEndDate()) == 0) {
+                addDatesToSet(bounds, queryDates, fieldIndexHole.getStartDate());
+                addDatesToSet(bounds, queryDates, fieldIndexHole.getEndDate());
+            }
+        }
+        
+        return queryDates;
+    }
+    
+    private void addDatesToSet(YearMonthDay.Bounds bounds, TreeSet<YearMonthDay> queryDates, String strDate) {
+        if (bounds.withinBounds(strDate))
+            queryDates.add(new YearMonthDay(strDate));
+    }
+    
+    @Override
+    public long maxRangesPerQueryPiece() {
+        return super.maxRangesPerQueryPiece();
+    }
+    
+    @Override
+    public void close(GenericQueryConfiguration config, Query settings) {
+        super.close(config, settings);
+    }
+    
+    @Override
+    public void setQueryIteratorClass(Class<? extends SortedKeyValueIterator<Key,Value>> clazz) {
+        super.setQueryIteratorClass(clazz);
+    }
+    
+    @Override
+    public Class<? extends SortedKeyValueIterator<Key,Value>> getQueryIteratorClass() {
+        return super.getQueryIteratorClass();
+    }
+    
+    @Override
+    public String getPlannedScript() {
+        return super.getPlannedScript();
+    }
+    
+    @Override
+    public DefaultQueryPlanner clone() {
+        return super.clone();
+    }
+    
+    @Override
+    public void setRules(Collection<PushDownRule> rules) {
+        super.setRules(rules);
+    }
+    
+    @Override
+    public Collection<PushDownRule> getRules() {
+        return super.getRules();
+    }
+    
+    @Override
+    public ASTJexlScript applyRules(ASTJexlScript queryTree, ScannerFactory scannerFactory, MetadataHelper metadataHelper, ShardQueryConfiguration config) {
+        return super.applyRules(queryTree, scannerFactory, metadataHelper, config);
+    }
+    
+    /**
+     * Calculate the FieldIndexHoles and add them to the ShardedQueryConfiguaration
+     *
+     * @param metadataHelper
+     * @param fieldToDatatypeMap
+     * @param config
+     */
+    public void calculateFieldIndexHoles(MetadataHelper metadataHelper, Multimap<String,Type<?>> fieldToDatatypeMap, ShardQueryConfiguration config)
```

Comment:
This method I would expect the metadataHelper to do for us.  Consider moving this into the metadata-utils. We should be able to generate FieldIndexHoles from the stored bitset directly instead of going through the intermediate YearMonthDay form.

Reply:
I don't like that transformation either.   I will get on that change ASAP.